ConfigMap generated by Helm contains duplicates

Running `helm template` with the parameters used for GKE creates a configmap which contains duplicate keys. This is likely overall a bug in the Helm charts and might _not_ be GKE specific. @twpayne discovered this while generating YAML files for GKE.

How to reproduce:
```console
$ helm template cilium cilium/cilium --version 1.9.0 \
  --namespace cilium \
  --set nodeinit.enabled=true \
  --set nodeinit.reconfigureKubelet=true \
  --set nodeinit.removeCbrBridge=true \
  --set cni.binPath=/home/kubernetes/bin \
  --set gke.enabled=true \
  --set ipam.mode=kubernetes \
  --set nativeRoutingCIDR=$NATIVE_CIDR > cilium-gke.yaml
$ pip install --user yamllint
...
$ yamllint -d relaxed --no-warnings cilium-gke.yaml
cilium-gke.yaml
  151:3     error    duplication of key "tunnel" in mapping  (key-duplicates)
  185:3     error    duplication of key "ipam" in mapping  (key-duplicates)

```

The `tunnel` key also contains conflicting values, which makes this potentially more dangerous:

```console
$ egrep 'tunnel|ipam' cilium-gke.yaml
  tunnel: vxlan
  ipam: "kubernetes"
  tunnel: "disabled"
  ipam: "kubernetes"

```

This also happens on Cilium 1.8.5.

--- a/install/kubernetes/cilium/templates/cilium-configmap.yaml
+++ b/install/kubernetes/cilium/templates/cilium-configmap.yaml
@@ -295,13 +295,6 @@ data:
   # container image names
   sidecar-istio-proxy-image: "{{ .Values.proxy.sidecarImageRegex }}"
 
-  # Encapsulation mode for communication between nodes
-  # Possible values:
-  #   - disabled
-  #   - vxlan (default)
-  #   - geneve
-  tunnel: {{ .Values.tunnel }}
-
   # Name of the cluster. Only relevant when building a mesh of clusters.
   cluster-name: {{ .Values.cluster.name }}
 
@@ -311,12 +304,17 @@ data:
   cluster-id: "{{ .Values.cluster.id }}"
 {{- end }}
 
+  # Encapsulation mode for communication between nodes
+  # Possible values:
+  #   - disabled
+  #   - vxlan (default)
+  #   - geneve
 {{- if .Values.gke.enabled }}
-
-  ipam: "kubernetes"
   tunnel: "disabled"
   enable-endpoint-routes: "true"
   enable-local-node-route: "false"
+{{- else }}
+  tunnel: {{ .Values.tunnel }}
 {{- end }}
 
 {{- if .Values.eni }}
@@ -612,9 +610,8 @@ data:
   # A space separated list of iptables chains to disable when installing feeder rules.
   disable-iptables-feeder-rules: {{ .Values.disableIptablesFeederRules | join " " | quote }}
 {{- end }}
-{{- if ne $ipam "hostscope" }}
   ipam: {{ $ipam | quote }}
-{{- end }}
+
 {{- if eq $ipam "cluster-pool" }}
 {{- if .Values.ipv4.enabled }}
   cluster-pool-ipv4-cidr: {{ .Values.ipam.operator.clusterPoolIPv4PodCIDR | quote }}

--- a/install/kubernetes/experimental-install.yaml
+++ b/install/kubernetes/experimental-install.yaml
@@ -119,18 +119,18 @@ data:
   # container image names
   sidecar-istio-proxy-image: "cilium/istio_proxy"
 
+  # Name of the cluster. Only relevant when building a mesh of clusters.
+  cluster-name: default
+  # Unique ID of the cluster. Must be unique across all conneted clusters and
+  # in the range of 1 and 255. Only relevant when building a mesh of clusters.
+  cluster-id: ""
+
   # Encapsulation mode for communication between nodes
   # Possible values:
   #   - disabled
   #   - vxlan (default)
   #   - geneve
   tunnel: vxlan
-
-  # Name of the cluster. Only relevant when building a mesh of clusters.
-  cluster-name: default
-  # Unique ID of the cluster. Must be unique across all conneted clusters and
-  # in the range of 1 and 255. Only relevant when building a mesh of clusters.
-  cluster-id: ""
   # Enables L7 proxy for L7 policy enforcement and visibility
   enable-l7-proxy: "true"
 

--- a/install/kubernetes/quick-install.yaml
+++ b/install/kubernetes/quick-install.yaml
@@ -98,18 +98,18 @@ data:
   # container image names
   sidecar-istio-proxy-image: "cilium/istio_proxy"
 
+  # Name of the cluster. Only relevant when building a mesh of clusters.
+  cluster-name: default
+  # Unique ID of the cluster. Must be unique across all conneted clusters and
+  # in the range of 1 and 255. Only relevant when building a mesh of clusters.
+  cluster-id: ""
+
   # Encapsulation mode for communication between nodes
   # Possible values:
   #   - disabled
   #   - vxlan (default)
   #   - geneve
   tunnel: vxlan
-
-  # Name of the cluster. Only relevant when building a mesh of clusters.
-  cluster-name: default
-  # Unique ID of the cluster. Must be unique across all conneted clusters and
-  # in the range of 1 and 255. Only relevant when building a mesh of clusters.
-  cluster-id: ""
   # Enables L7 proxy for L7 policy enforcement and visibility
   enable-l7-proxy: "true"
 

